public/index.html

```diff
@@ -115,6 +132,50 @@ <h2>PayJoin here to open these channels</h2>
 				channels.removeChild(channels.lastElementChild);
 			}
 
+			async function get_recommended_channels(){
+				document.getElementsByClassName('recommendation-button-wrapper')[0].classList.add('invisible')
+				try {
+					recommendations = await (await fetch('/recommend', {
+						method: 'GET',
+					})).json()
+
+					for (var node_type in recommendations) {
+						let text = `
+						<div>
+							<h3>${node_type}</h3>
+							<small>a well connected node that is reliable for payments</small>
+							<br>
+							<br>
+							<ul>
+								<li>${recommendations[node_type].alias}</li>
+								<li>&nbsp;Active Capacity:&nbsp;₿${recommendations[node_type].capacity / SATS_PER_BTC} | &nbsp;Channel Count:&nbsp;${recommendations[node_type].active_channel_count}</li> 
+							</ul>
+							<div>
+								<button class="colored-button" type="button" onclick="add_recommended_node('${node_type}')">Use</button>	
+							</div>
+						</div>
+						<br>
+						`;
+						document.getElementById("recommendations").insertAdjacentHTML("beforeend", text);
+					}
+				} catch (error) {
+					console.error(error)
+					alert("Sorry, can't connect to node sources right now.")
+				}
+			}
+
+			function add_recommended_node(type = 'routing_node') {
+				const node = recommendations[type];
+				if (!node) {
+					alert("Something went wrong trying to add node to channel list. Please try again or refresh.");
+					return;
+				}
+				const pubkey = node.public_key;
+				const socket = node.sockets.includes(',') ? node.sockets.split(',')[0] : node.sockets;
```

Comment:
this is not view logic, this is business logic 🚩

Reply:
Whats the suggestion? I think it's fine for the client to have rich business logic and a dumb presentation view. - all in one application That's how modern web apps are architected. (MVVM)

Reply:
Yes, and in our case this html file is the view and the http controller would be the viewmodel. This JS makes assumptions about the format of node.sockets. Instead, pass the whole p2paddress as a string, validated in rust, so the js doesn't need to know anything about the form of pubkey or socket or p2paddress.